### PR TITLE
WIP: Serve LFS/attachment with http.ServeContent to Support Range-Request

### PR DESCRIPTION
**Pull request by 6543**
take #18448 and finish it ... 

```
The `ServeData` function now can't handle [Range Request](https://developer.mozilla.org/en-US/docs/Web/HTTP/Range_requests).
Due to the complexity of this part of HTTP protocol, it is difficult to fully implement it and requires a lot of testing.
The official module `http.ServeContent` has this implementation, and we can replace with it directly.
It can handle `If-Match, If-Unmodified-Since, If-None-Match, If-Modified-Since, If-Range` and more, perfectly!
```

### CODE REVIEW

File: modules/typesniffer/typesniffer.go [hunk 1]
```diff
@@ -70,6 +72,11 @@ func (ct SniffedType) IsRepresentableAsText() bool {
 	return ct.IsText() || ct.IsSvgImage()
 }
 
+// Mime return the mime
+func (ct SniffedType) Mime() string {
+	return strings.Split(ct.contentType, ";")[0]
+}
+
```
Comment (silverwind) [inline modules/typesniffer/typesniffer.go:79]: Note to self: this is a better version than I have in https://github.com/go-gitea/gitea/pull/20464, will incorporate there.

Comment (6543) [reply modules/typesniffer/typesniffer.go:79]: we should use SplitN so go can skip split after first ; ... just some tiny optimization nits

Comment (6543) [reply modules/typesniffer/typesniffer.go:79]: @silverwind I'm going to delete the mime stuff as it's unrelated to the pull topic! - so feel free to pick it for your pull :)

Comment (silverwind) [reply modules/typesniffer/typesniffer.go:79]: sure, for reference, code was:

```golang
func (ct SniffedType) Mime() string {
	return strings.Split(ct.contentType, ";")[0]
}

Comment (silverwind) [reply modules/typesniffer/typesniffer.go:79]: If we are talking about micro-optimization, `str[:strings.Index(str, ";")]` should be even faster than SplitN I think :)

Edit: Done, added it as `GetMimeType` in https://github.com/go-gitea/gitea/pull/20484.

Comment (silverwind) [inline modules/typesniffer/typesniffer.go:79]:
```suggestion
```

Delete as discussed, it's unused.

File: modules/setting/mime_type_map.go [hunk 2]
```diff
@@ -21,6 +26,10 @@ func newMimeTypeMap() {
 	m := make(map[string]string, len(keys))
 	for _, key := range keys {
 		m[strings.ToLower(key.Name())] = key.Value()
+		err := mime.AddExtensionType(key.Name(), key.Value())
+		if err != nil {
+			log.Warn("mime.AddExtensionType(%s,%s): %v", key.Name(), key.Value(), err)
+		}
```
Comment (silverwind) [inline modules/setting/mime_type_map.go:32]: This is also unused I think.

File: routers/common/repo.go [hunk 3]
```diff
@@ -38,49 +40,75 @@ func ServeBlob(ctx *context.Context, blob *git.Blob, lastModified time.Time) err
...
+	_, err = ctx.Resp.Write(buf)
+	if err != nil {
+		return err
 	}
-	name = path.Base(name)
+	_, err = io.Copy(ctx.Resp, dataRc)
+	return err
+}
 
+func setCommonHeaders(ctx *context.Context, name string, data interface{}) error {
```
Comment (silverwind) [inline routers/common/repo.go:71]: Not sure this additional function is of any benefit. Cache header can be set like in https://github.com/go-gitea/gitea/pull/20484 via `httpcache` module.

**Comment by wxiaoguang** [comment]
->  Make repository response support HTTP range request #24592

**Pull request closed without merging**